docs/guides/field-extensions.md

```diff
@@ -0,0 +1,170 @@
+## title: Field extensions
+
+# Field extensions
+
+Field extensions are a great way to implement reusable logic such as permissions or
+pagination outside your resolvers. They wrap the underlying resolver and are able to
+modify the field and all arguments passed to the resolver.
+
+<Note>
+
+The following examples only cover sync execution. To use extensions in async contexts,
+please have a look at [Async Extensions and Resolvers](#Async Extensions and Resolvers)
+
+</Note>
+
+```python
+import strawberry
+from strawberry.extensions import FieldExtension
+
+
+class UpperCaseExtension(FieldExtension):
+    def resolve(self, next_: Callable[..., Any], source: Any, info: Info, **kwargs):
+        result = next_(source, info, **kwargs)
+        return str(result).upper()
+
+
+@strawberry.type
+class Query:
+    @strawberry.field(extensions=[UpperCaseExtension()])
+    def string(self) -> str:
+        return "This is a test!!"
+```
+
+In this example, the `UpperCaseExtension` wraps the resolver of the `string`
+field (`next`) and modifies the resulting string to be uppercase.
+The extension will be called instead of the resolver and receives
+the resolver function as the `next` argument. Therefore, it is important
+to not modify any arguments that are passed to `next` in an incompatible way.
+
+```gql+json
+query {
+    string
+}
+---
+{
+  "string": "THIS IS A TEST!!"
+}
+```
+
+## Modifying the field
```

Comment:
I think it's worth adding a warning here about `StrawberryField` since it's the first time that we're exposing it publicly and it's got some weird edges. We also (in the future) need ways to easily change the type of the field and modify its arguments.

Reply:
Do you have any pointers on what could be worthwhile to warn?

Reply:
Maybe just something like:

Warning: the StrawberryField API is not stable and might change in the future without warning.

Reply:
Done ☺️